src/app/services/baw-api/account.service.spec.ts

```diff
@@ -8,214 +8,378 @@ import { Subject } from "rxjs";
-              rolesMaskNames: ["user"],
-              lastSeenAt: "1970-01-01T00:00:00.000"
-            })
-          );
+          subject.next(output);
           subject.complete();
         }, 50);
 
         return subject;
-      }
-    );
-
-    service.show(5).subscribe(
-      (user: User) => {
-        expect(user).toBeTruthy();
-        expect(user).toEqual(
-          new User({
-            id: 5,
-            userName: "username",
-            rolesMask: 3,
-            rolesMaskNames: ["user"],
-            lastSeenAt: "1970-01-01T00:00:00.000"
-          })
-        );
-      },
-      () => {
-        expect(true).toBeFalsy("Service should not return an error");
-      }
-    );
-
-    tick(100);
-  }));
-
-  it("should handle error", fakeAsync(() => {
-    spyOn<any>(service, "show").and.callFake(
-      (path: string, filters: Filters) => {
-        expect(path).toBe("/user_accounts/1");
-        expect(filters).toEqual(null);
-        const subject = new Subject<User>();
+      });
+    }
 
-        setTimeout(() => {
-          subject.error({
-            status: 401,
-            message: "Unauthorized"
-          } as ApiErrorDetails);
-        }, 50);
+    it("should handle user input", fakeAsync(() => {
+      const model = new User({
+        id: 1,
+        userName: "username"
+      });
 
-        return subject;
-      }
-    );
-
-    service.show(1).subscribe(
-      () => {
-        expect(true).toBeFalsy("Service should not return data");
-      },
-      (err: ApiErrorDetails) => {
+      createSuccess("/user_accounts/1", model);
+
+      service.show(model).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
+
+    it("should handle user id input", fakeAsync(() => {
+      const model = new User({
+        id: 1,
+        userName: "username"
+      });
+
+      createSuccess("/user_accounts/1", model);
+
+      service.show(1).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
+
+    it("should handle response with random user model id", fakeAsync(() => {
+      const model = new User({
+        id: 5,
+        userName: "username"
+      });
+
+      createSuccess("/user_accounts/5", model);
+
+      service.show(model).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
+
+    it("should handle response with random user id", fakeAsync(() => {
+      const model = new User({
+        id: 5,
+        userName: "username"
+      });
+
+      createSuccess("/user_accounts/5", model);
+
+      service.show(5).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
```

Comment:
these tests seem redundant - what are they testing?

Reply:
The difference is that "should handle response with random user model id" uses the User model to get the id, whereas "should handle response with random user id" passes the id directly. Its mostly a check that the id is being read correctly from both types of inputs using an id value which is not 1 (which i use as a standard placeholder id)

Reply:
Ah fair enough.